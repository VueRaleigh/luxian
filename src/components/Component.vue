<template>
    <div>
        <h1>Test component</h1>
        <label for="b1">Click to fetch results</label>
        <button id="b1" @click="fetchResults" />
        <p>this is a test component with a "{{ value }}"</p>
    </div>
</template>

<script>

import axios from 'axios';

export default {
    data() {
        return {
            value: ''
        };
    },
    methods: {
      async fetchResults() {
        const response = await axios.get('mock/service')
        this.value = response.data
      }
    }
}
</script>
