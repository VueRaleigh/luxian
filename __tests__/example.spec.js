import { shallowMount } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import { mount } from '@vue/test-utils'
import Component from '@/components/Component'

jest.mock('axios')

describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Component)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(Component)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('fetches async when a button is clicked', async () => {
    const wrapper = shallowMount(Component)
    wrapper.find('button').trigger('click')

    await flushPromises();

    expect(wrapper.vm.value).toBe('value')
  })
})
