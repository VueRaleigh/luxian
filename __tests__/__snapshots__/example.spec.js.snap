// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div>
  <h1>
    Test component
  </h1>
   
  <label
    for="b1"
  >
    Click to fetch results
  </label>
   
  <button
    id="b1"
  />
   
  <p>
    this is a test component with a ""
  </p>
</div>
`;
